### DISCUSSION TITLE
Map a stream to another type

### DISCUSSION THREAD

**Discussion by ev94** [hyperium/tonic#1079] [Q&A] [answered]
Hi,
Thx for the great work. We ve been using tonic in prod for several months and everything is fine.
I've a dev question. I would like to map a Stream to another type but it doesn't work:

```
use futures::stream::StreamExt;

let grpc_req: GrpcLiveEventsRequest = req.into();
let grpc_stream = self.es_service.live_events(grpc_req).await?.into_inner(); 
let stream: Streaming<String> = grpc_stream
    .map(|_| "test".to_string())
    .into_inner();
```

stream is still of the previous type `Streaming<Events>` instead of `Streaming<String>`
What am I missing? thx

**Comment by davidpdrsn** [comment, marked as answer]
You don't call `into_inner()` on a `Streaming`. [It doesn't have that method](https://docs.rs/tonic/latest/tonic/struct.Streaming.html).

You can just call `streaming.map(...)` like you're doing, where `map` comes from [`StreamExt`](https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html#method.map)

**Reply by ev94** [reply]
Thx for your quick reply.
I don't get it. This code compiles.
The first into_inner turns `Response<Streaming<Events>>` into `Streaming<Events>`
The second one is on https://docs.rs/futures/latest/futures/stream/struct.Map.html

**Reply by ev94** [reply]
Understood the second into_inner returns the underlying stream
